superset/utils/pandas_postprocessing/contribution.py

```diff
@@ -71,5 +73,7 @@ def contribution(
     numeric_df = numeric_df[columns]
     axis = 0 if orientation == PostProcessingContributionOrientation.COLUMN else 1
     numeric_df = numeric_df / numeric_df.values.sum(axis=axis, keepdims=True)
+    # replace infinity and nan with 0 in dataframe
+    numeric_df.replace(to_replace=[np.Inf, -np.Inf, np.nan], value=0, inplace=True)
```

Comment:
There is a Dateframe like above:

```
    __timestamp      a      b     c
0 2020-07-16 14:49:00  1.0  1.0 NaN
1 2020-07-16 14:50:00  3.0  9.0 NaN
2 2020-07-16 14:51:00  NaN  NaN NaN
```

The df will be transformation like following after run `contribution`:

```
contribution(
    df, orientation=PostProcessingContributionOrientation.ROW,
)

__timestamp     a     b    c
0 2020-07-16 14:49:00  0.50  0.50  0.0
1 2020-07-16 14:50:00  0.25  0.75  0.0
2 2020-07-16 14:51:00  0.00  0.00  0.0
```

In the other hand, If we didn't replace the `nan` and `inf` values, the row of `2020-07-16` will be filled out nan in every cell, **but**  cell of `row: 2020-07-16 column: c ` have be filled in **0**.

```
           __timestamp     a     b    c
0 2020-07-16 14:49:00  0.50  0.50  0.0
1 2020-07-16 14:50:00  0.25  0.75  0.0
2 2020-07-16 14:51:00   NaN   NaN  NaN
```

Reply:
Thanks for the explanation! This is mostly a nit, but please hear me out 😆 I agree that we need to fill nulls with zeros before doing the contribution calculation, but strictly mathematically speaking, after we've done the contribution calculation, I think leaving the infinite values as null seems more appropriate. In the example, I feel this should be the correct result for ROW level contribution, as there strictly speaking isn't anything to contribute to:

```
           __timestamp     a     b    c
0 2020-07-16 14:49:00  0.50  0.50  0.0
1 2020-07-16 14:50:00  0.25  0.75  0.0
2 2020-07-16 14:51:00   NaN   NaN  NaN
```

Reply:
Oh! Thanks for the explanation. It makes sense. Updaing......